app/src/main/java/org/gdg/frisbee/android/common/GdgActivity.java

```diff
@@ -114,14 +114,16 @@ public void onCreate(Bundle savedInstanceState) {
     }
 
     protected void createGoogleApiClient() {
-        mGoogleApiClient = new GoogleApiClient.Builder(this)
-                .addApi(Plus.API)
-                .addApi(Games.API)
-                .addApi(AppStateManager.API)
-                .addScope(Plus.SCOPE_PLUS_LOGIN)
-                .addScope(Plus.SCOPE_PLUS_PROFILE)
-                .addScope(Games.SCOPE_GAMES)
-                .build();
+        GoogleApiClient.Builder builder = new GoogleApiClient.Builder(this)
+                .addApiIfAvailable(AppStateManager.API);
+        if (PrefUtils.isSignedIn(this)) {
+            builder.addApiIfAvailable(Plus.API, Plus.SCOPE_PLUS_LOGIN, Plus.SCOPE_PLUS_PROFILE)
```

Comment:
UpdateUserProfilePicture should be updated, shouldn't it?

What should happen if play services are not available? The 'if applicable'
was added to handle this case better. Not connecting at all would result in
poorer experience for signed out users.
On 16 Dec 2015 14:05, "Said Tahsin Dane" <notifications@github.com> wrote:

> In app/src/main/java/org/gdg/frisbee/android/common/GdgActivity.java
> <https://github.com/gdg-x/frisbee/pull/501#discussion_r47772850>:
>
> > @@ -114,14 +114,16 @@ public void onCreate(Bundle savedInstanceState) {
> >      }
> >
> >      protected void createGoogleApiClient() {
> > -        mGoogleApiClient = new GoogleApiClient.Builder(this)
> > -                .addApi(Plus.API)
> > -                .addApi(Games.API)
> > -                .addApi(AppStateManager.API)
> > -                .addScope(Plus.SCOPE_PLUS_LOGIN)
> > -                .addScope(Plus.SCOPE_PLUS_PROFILE)
> > -                .addScope(Games.SCOPE_GAMES)
> > -                .build();
> > +        GoogleApiClient.Builder builder = new GoogleApiClient.Builder(this)
> > +                .addApiIfAvailable(AppStateManager.API);
> > +        if (PrefUtils.isSignedIn(this)) {
> > +            builder.addApiIfAvailable(Plus.API, Plus.SCOPE_PLUS_LOGIN, Plus.SCOPE_PLUS_PROFILE)
>
> I think this part makes problems. I am getting this crash:
>
> java.lang.IllegalStateException: GoogleApiClient has an optional Plus.API and is not connected to Plus. Use GoogleApiClient.hasConnectedApi(Plus.API) to guard this call.
>                                                                                    at com.google.android.gms.plus.Plus.zzf(Unknown Source)
>                                                                                    at com.google.android.gms.internal.zzpr.getCurrentPerson(Unknown Source)
>                                                                                    at org.gdg.frisbee.android.common.GdgNavDrawerActivity.updateUserPicture(GdgNavDrawerActivity.java:350)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gdg-x/frisbee/pull/501/files#r47772850>.
>